__init__.py

```diff
@@ -110,19 +115,16 @@ def find_station(self, search_term):
             # Only look at outlines that are of type=audio and item=station
             if (entry.getAttribute("type") == "audio") and (entry.getAttribute("item") == "station"):
                 if (entry.getAttribute("key") != "unavailable"):
-                    # stop the current stream if we have one running
-                    if (self.audio_state == "playing"):
-                        self.stop()
                     # Ignore entries that are marked as unavailable
                     self.mpeg_url = entry.getAttribute("URL")
                     self.station_name = entry.getAttribute("text")
                     # this URL will return audio/x-mpegurl data. This is just a list of URLs to the real streams
                     self.stream_url = self.get_stream_url(self.mpeg_url)
-                    self.audio_state = "playing"
+                    LOG.debug("Found stream URL: " + self.stream_url)
+                    self.audio_service = AudioService(self.bus)
```

Comment:
suggestion: move this into initialize, you can reuse the audio service object this way